### ISSUE TITLE
Remove Filesystem's argv_0 parameter

### ISSUE THREAD

**Issue by cugone** [lairworks/nas2d-core#390]
The recent changes to Filesystem Update use OS-specific calls to get the path to the exe. As such, the argv_0 argument is no longer needed.

**Comment by DanRStevens** [comment]
I'm not convinced it is no longer needed. Use of this parameter depended on the platform. On Windows it was not needed. It may still be needed on Linux. I'm less certain about MacOS, though it may (optionally?) function similar to Linux.

**Comment by cugone** [comment]
The system now uses OS calls exclusively for all platforms the variable literally isn't used at all. Therefore, it's not needed.

**Comment by DanRStevens** [comment]
Came across this, which right at the bottom has code for finding the executable path:
https://www.doxsey.net/blog/a-response-to-hello-world

Interesting how it uses `/proc/self` on Linux. That may be a good indication it is a reliable way to get the path. Would love to find some documentation that addresses how portable it is. Curiously the code uses a different path for BSD derived operating systems.

As a side note, this makes me curious how many and what operating systems are supported by PhysFS.

Related, which do we care about. We currently target 3 main OSes. Though the decision is largely because only those 3 ever came to mind.

**Comment by cugone** [comment]
I could have it try to use the current implementation and if it doesn't exist, fall back to `/proc/self/exe` and follow its symlinks all the way to a real file.

**Comment by DanRStevens** [comment]
This is currently blocked by:
- Issue #212
- Replacing `Filesystem`'s implementation so it doesn't use PhysFS (What PR #243 is attempting)
